samples/CMakeLists.txt

```diff
@@ -117,6 +117,16 @@ if (CONFIG_NCS_INCLUDE_RPMSG_CHILD_IMAGE OR CONFIG_NCS_SAMPLE_EMPTY_NET_CORE_CHI
       )
     endif()
 
+    if (CONFIG_BT_MESH)
+      # Inject this configuration from parent image to enable
+      # necessary amount of advertising sets in the controller on the
+      # network core required by the Bluetooth Mesh
+      add_overlay_config(
+        ${CHILD_IMAGE_NAME}
+        "${NRF_DIR}/subsys/bluetooth/mesh/hci_rpmsg_child_image_overlay.conf"
```

Comment:
The `hci_rpmsg` image is added to the compilation in `samples/CMakeLists.txt` file if `CONFIG_NCS_INCLUDE_RPMSG_CHILD_IMAGE` is enabled. This option is enabled for nRF53 (https://github.com/nrfconnect/sdk-nrf/blob/6a54e96a7244c17a8d7ede6d290defd269211af2/samples/Kconfig#L33) because `CONFIG_BT_RPMSG` is automagically selected here: https://github.com/nrfconnect/sdk-zephyr/blob/74c7bca86e6a7d590333681eef90e159d79bcfe8/drivers/bluetooth/hci/Kconfig#L11-L40. So my guess is that this will work not only for samples.

The other thing is that this configuration won't show up as a default configuration in the net core, because there is no `CONFIG_BT_MESH` in the net core configuration and we can't add it there as it depends on `CONFIG_BT_HCI_HOST` option which is replaced by `CONFIG_BT_HCI_RAW` on the net core. I explained this in details in the comment to NCSDK-18316 task.

Re-creating `child_image/hci_rpmsg.conf` file here and specifying a new value for `CONFIG_BT_EXT_ADV_MAX_ADV_SET` will override the configuration of this overlay file.